### PR TITLE
Implement step 1.2 of event handler compilation

### LINKED ISSUE #13363: Check whether the document has scripting enabled when compiling event handlers
There's a TODO about checking whether scripting is enabled. We can now do that using `document.is_scripting_enabled()`, so we should.

Code: `components/script/dom/eventtarget.rs` (`get_compiled_event_handler`)

I haven't thought of a way to test this, so no need to spend time figuring that out.

### PR DESCRIPTION
**Pull request by phrohdoh**
Closes #13363

---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes fix #13363

- [X] These changes do not require tests because we haven't discussed _how_ to test this

---

This change is [Reviewable](https://reviewable.io/reviews/servo/servo/13365)

### CODE REVIEW

**Comment by KiChjang** [comment]
@bors-servo r+

Thanks!

File: components/script/dom/eventtarget.rs [hunk 1]
```diff
@@ -375,7 +375,10 @@ impl EventTarget {
             None => self.downcast::<Window>().unwrap().Document(),
         };
 
-        // TODO step 1.2 (browsing context/scripting enabled)
+        // Step 1.2
+        if document.browsing_context().is_none() || !document.is_scripting_enabled() {
```
Comment (Ms2ger) [inline components/script/dom/eventtarget.rs]: Those two checks are (currently) completely equivalent. Let's just call `is_scripting_enabled()`; that should always include the browsing context check.

**Comment by jdm** [comment]
Aha, I totally forgot about parseFromString. Excellent testing solution.

**Comment by Phrohdoh** [comment]
Could I get an example of something with an event handler attribute?

**Comment by jdm** [comment]
`<div onclick="js code goes here">`

**Comment by Phrohdoh** [comment]
Oh, thanks! I did not realize that is what we were discussing.
I'll work on this later today.

**Comment by Phrohdoh** [comment]
I apologize for being so late with this, I have been quite busy with other parts of life recently.
I will try to tend to this tomorrow, failing that it should probably be opened back up for someone else to grab. I don't want to hinder something like this.

**Comment by Phrohdoh** [comment]
```
12:10:27 PM ~jdm Phrohdoh: for the second question, we have have the handler set property on the document object and have the calling code check that the property doesn't exist 
12:10:43 PM ~jdm Phrohdoh: for checking the event, that's event.defaultPrevented iirc 
12:11:14 PM ~jdm Phrohdoh: for window.onerror, add an event listener for the error event on window and make the test fail if it's invoked 
12:11:38 PM ~jdm (ie. have a onErrorInvoked = false, set it to true in the handler, and assert that it's false before finishing the test 
12:13:09 PM ~jdm Phrohdoh: for where it should live, tests/wpt/web-platform-tests/webappapis/scripting/events/ 
12:13:46 PM ~jdm Phrohdoh: since the specification text being implemented exists in the web app apis part of the spec, under scripting and events 
12:13:57 PM Phrohdoh What should the filename be? 
12:14:40 PM ~jdm Phrohdoh: uncompiled_event_handler_with_scripting_disabled.html 
```

**Comment by KiChjang** [comment]
@Phrohdoh Are you still working on this?

**Comment by Phrohdoh** [comment]
Sorry, I do not have time right now to continue work on this, so no.

**Pull request closed without merging**